src/accelerate/accelerator.py

```diff
@@ -2579,7 +2581,14 @@ def save_model(
         os.makedirs(save_directory, exist_ok=True)
 
         # get the state_dict of the model
-        state_dict = self.get_state_dict(model)
+        if (
+            hasattr(model, "_hf_hook")
+            and isinstance(model._hf_hook, AlignDevicesHook)
+            and True in {module._hf_hook.offload for _, module in model.named_modules() if hasattr(module, "_hf_hook")}
+        ):
```

Comment:
```suggestion
            and any(module._hf_hook.offload for module in model.modules() if hasattr(module, "_hf_hook") and isinstance(module._hf_hook, AlignDevicesHook))
        ):
```
I've simplified a bit. Also only `AlignDevicesHook` have the offload attribute.